test/Sdk.Generator.Tests/FunctionMetadataProviderGeneratorTests/ServiceBustTests.cs

```diff
@@ -0,0 +1,149 @@
+            {
+                // test generating function metadata for a simple HttpTrigger
+                string inputCode = """
+                using System;
+                using System.Threading.Tasks;
+                using Azure.Messaging.ServiceBus;
+                using Microsoft.Azure.Functions.Worker;
+                using Microsoft.Extensions.Logging;
+
+                namespace SampleApp
+                {
+                    public static class TimerFunction
+                    {
+                        [Function("ServiceBusFunction")]
+                        public static void Run([ServiceBusTrigger("queue", Connection = "ServiceBusConnection", IsSessionsEnabled = true)]
+                        ServiceBusReceivedMessage message,
+                        ServiceBusMessageActions messageActions)
+                        {
+                            throw new NotImplementedException();
+                        }
+                    }
+                }
+                """;
+
+
+                string expectedGeneratedFileName = $"GeneratedFunctionMetadataProvider.g.cs";
+                string expectedOutput = """
+                // <auto-generated/>
+                using System;
+                using System.Collections.Generic;
+                using System.Collections.Immutable;
+                using System.Text.Json;
+                using System.Threading.Tasks;
+                using Microsoft.Azure.Functions.Worker;
+                using Microsoft.Azure.Functions.Worker.Core.FunctionMetadata;
+                using Microsoft.Extensions.DependencyInjection;
+                using Microsoft.Extensions.Hosting;
+                
+                namespace TestProject
+                {
+                    /// <summary>
+                    /// Custom <see cref="IFunctionMetadataProvider"/> implementation that returns function metadata definitions for the current worker."/>
+                    /// </summary>
+                    [global::System.ComponentModel.EditorBrowsableAttribute(global::System.ComponentModel.EditorBrowsableState.Never)]
+                    [global::System.Runtime.CompilerServices.CompilerGeneratedAttribute()]
+                    public class GeneratedFunctionMetadataProvider : IFunctionMetadataProvider
+                    {
+                        /// <inheritdoc/>
+                        public Task<ImmutableArray<IFunctionMetadata>> GetFunctionMetadataAsync(string directory)
+                        {
+                            var metadataList = new List<IFunctionMetadata>();
+                            var Function0RawBindings = new List<string>();
+                            Function0RawBindings.Add(@"{""name"":""message"",""type"":""serviceBusTrigger"",""direction"":""In"",""properties"":{""supportsDeferredBinding"":""True""},""queueName"":""queue"",""connection"":""ServiceBusConnection"",""isSessionsEnabled"":true,""cardinality"":""One""}");                
```

Comment:
NOTE: Inside properties object, "supportsDeferredBinding" is kept as `TrueString` value because this is the same behavior as is implemented in `functions.metadata` [here](https://github.com/Azure/azure-functions-dotnet-worker/blob/5bb9fe9fb8d7c03f630649cd33bcbf2a2078db86/test/SdkE2ETests/Contents/functions.metadata#L56).

Reply:
@kshyju this answers your other comment